packaging/mingw-w64/configure-build-msys2-mingw-static.sh

```diff
@@ -0,0 +1,59 @@
+#!/bin/bash
+
+set -e
+
+cmake \
+    -G "MinGW Makefiles" \
+    -D CMAKE_INSTALL_PREFIX="$PWD/dest/" \
+    -D WITHOUT_WIN32_CONFIG=ON  \
+    -D RDKAFKA_BUILD_EXAMPLES=ON \
+    -D RDKAFKA_BUILD_TESTS=ON \
+    -D WITH_LIBDL=OFF \
+    -D WITH_PLUGINS=OFF \
```

Comment:
Why is this OFF? Plugins are supported on Windows

Reply:
We don't need LIBDL on windows, plugins are loaded with LoadLibrary.
So keep WITH_LIBDL=OFF but set WITH_PLUGINS=on

Reply:
Updated CMakeLists.txt to default WITH_PLUGINS ON on Windows. Also, there was a typo in src/CMakeLists.txt that caused WITH_PLUGINS to not work properly on Windows builds.